File: frontend/interfaces/ChargingStation.ts
export default interface ChargingStation {
  coords: {
    lat: number;
    lng: number;
  };
  name: string;
  address: string;
  id: string;
  chargingPorts: number;
  maximumPower: number;
  type: string; // AC or DC
  pricePerKWh: number;
}

File: backend/model/ChargingStation.py
from pydantic import BaseModel
from typing import Optional
class Coordinates(BaseModel):
    lat: float
    lng: float

class ChargingStation(BaseModel):
    coords: Coordinates
    name: str
    address: str
    id: str
    chargingPorts: int
    maximumPower: float  
    type: str            
    pricePerKWh: float   
    access_node_id: Optional[int] = None
